inc/namespace.php

```diff
@@ -39,8 +63,7 @@ function register_settings() {
  * The basic auth override setting.
  */
 function basic_auth_setting_callback() {
-	$hm_dev  = defined( 'HM_DEV' ) ? absint( HM_DEV ) : false;
-	$checked = get_option( 'hm-basic-auth' ) ?: $hm_dev;
+	$checked = get_option( 'hm-basic-auth' ) ?: is_development_environment();
```

Comment:
This doesn't seem to allow a way to turn off basic auth on dev/staging environments, because a false value for the `hm-basic-auth` option will coalesce into the return value of `is_development_environment()`. Is this intentional?

Reply:
Oh, wait, never mind. I read below and saw that you're storing false values as "off" instead of false. So the first part of this expression will evaluate properly. The problem I think might be that `is_development_environment()` returns a boolean and you're expecting a string "on" or "off".

Reply:
This change just simplifies these two lines of code into one line and refactors (a bit) since we're storing `"off"` and `"on"` strings.

`$hm_dev  = defined( 'HM_DEV' ) ? absint( HM_DEV ) : false;`

This can be simplified by simply using `is_development_environment()` which already performs this check (as well as all the others that we need). It returns a `false` value if we're not in a development environment so there's no need to store it in a variable, we can just use the output of the function.

`$checked = get_option( 'hm-basic-auth' ) ?: is_development_environment();`

Check the value of the `hm-basic-auth` option. If it's false or doesn't exist, default to the `is_development_environment()` value. Note, _this only affects the **checked** status_ and since we're explicitly storing `on` and `off` in the database, the only time `get_option` would return empty or false (and, therefore use `is_development_environment` instead) is when _no setting_ has been saved.

I've doublechecked this a couple times to make sure it's still working as expected and it seems to be all good. However I can add some inline comments since it does look counter-intuitive.

Reply:
Right, OK. So the only reason the checkbox should be checked is if the option value is set to "on"? I guess in that case it's doing what's intended. I still don't see why the `is_development_environment()` check is in there. Why would you be checking that value if it's not going to be used?

Reply:
> Why would you be checking that value if it's not going to be used?

If the option is not set (e.g. initial install) we're displaying whatever the environment determines the status should be. Otherwise it would appear to be _disabled_ because there's no value (yet) in the database. Unless I'm overthinking things?

Reply:
I guess to clarify what I'm seeing:

If no value is set for this option and we're on a development site, this line will evaluate to `true`. But then in the checkbox you have `<?php checked( $checked, 'on' ); ?>`, which will not output anything if `$checked` is `true`.

Reply:
Right. I wasn't seeing it because I was toggling the option. 👍